Plugins/NativeEngine/Source/NativeEngine.h

```diff
@@ -43,11 +43,26 @@ namespace Babylon
     struct ProgramData final
     {
         ProgramData() = default;
-        ProgramData(ProgramData&& other) = delete;
         ProgramData(const ProgramData&) = delete;
-        ProgramData& operator=(ProgramData&& other) = delete;
         ProgramData& operator=(const ProgramData& other) = delete;
 
+        ProgramData(ProgramData&& other) noexcept
+        {
+            (*this) = std::move(other);
+        }
```

Comment:
Not using constructor initialization will cause the fields to be default constructed for no reason.

```suggestion
        ProgramData(ProgramData&& other) noexcept :
            Handle{other.Handle},
            Uniforms{std::move(other.Uniforms)},
            UniformToIndex{std::move(other.UniformToIndex)},
            UniformInfos{std::move(other.UniformInfos)},
            VertexAttributeLocations{std::move(other.VertexAttributeLocations)},
        {
            other.Handle = BGFX_INVALID_HANDLE;
        }
```

I also think we should remove the Disposed flag. It's the same as Handle being invalid.

Reply:
Currently, `Dispose` checks whether `Handle` is valid, but doesn't actually change it when disposing the BGFX resource. I assume that leaves it in a state where `Disposed` is `true` but `Handle` is still valid. Should `Dispose` set `Handle` to `bgfx::kInvalidHandle` when successfully disposing the BGFX resource? Or might there be a reason why we want to remember which program handle was used even after it is disposed?

Reply:
I don't think we need to keep the Handle with the previous value. It might be useful for debugging at some point, but I've never heard of anyone trying to use it before.